.github/workflows/run-tests.yml

```diff
@@ -343,7 +343,7 @@ jobs:
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
           registry: docker.pkg.github.com
-          image_name: qgis3-qt${{ matrix.qt-version }}-build-deps-bin-only
+          image_name: qgis3-build-deps-${{ matrix.distro-version }}
```

Comment:
that won't work if you use the same name than in the first job, no?

what about qgis3-test-run-${{ matrix.distro-version }}-${{ matrix.test-batch }}

Reply:
Everything will work if I use the same name `qgis3-build-deps-${{ matrix.distro-version }}` as in the first job!

But in this case, if the log file is to be believed, you would have to download the 4.95 GB file from the docker repository instead of the 2.14 GB required for Oracle tests.

https://github.com/qgis/QGIS/runs/3269842556 and open step `Print disk space`

```
docker.pkg.github.com/qgis/qgis/qgis3-build-deps-20.04-stages   3                                  fc5fe6b6daa7   32 hours ago         4.95GB
docker.pkg.github.com/qgis/qgis/qgis3-build-deps-20.04-stages   2                                  4f2580b540f3   32 hours ago         3.7GB
my_awesome_image                                                latest                             4eb3d7207cdf   32 hours ago         2.14GB
qgis3-build-deps-binary-image                                   latest                             4eb3d7207cdf   32 hours ago         2.14GB
docker.pkg.github.com/qgis/qgis/qgis3-build-deps-20.04-stages   1                                  4eb3d7207cdf   32 hours ago         2.14GB
docker.pkg.github.com/qgis/qgis/qgis3-build-deps-20.04          latest                             4eb3d7207cdf   32 hours ago         2.14GB
```

Reply:
by not working I meant it is sub-optimal, as you are either downloading too much, uploading for nothing.
i.e. in the current config in this PR, the test jobs will update 4 times with the same image name, overwriting each time the other one.

is there any reason not to use the `test-batch` in the image name?

Reply:
> is there any reason not to use the `test-batch` in the image name?

https://github.com/qgis/QGIS/actions/runs/1104824050 :)

Reply:
another bad idea, push the docker cache as an artifact, something like this :)
https://github.com/MorriganR/QGIS/actions/runs/1134543505